framework/src/main/java/org/checkerframework/common/accumulation/AccumulationVisitor.java

```diff
@@ -0,0 +1,41 @@
+package org.checkerframework.common.accumulation;
+
+import com.sun.source.tree.AnnotationTree;
+import javax.lang.model.element.AnnotationMirror;
+import javax.tools.Diagnostic;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.basetype.BaseTypeVisitor;
+import org.checkerframework.framework.source.DiagMessage;
+import org.checkerframework.javacutil.TreeUtils;
+
+/**
+ * The visitor for an accumulation checker. Issues predicate.invalid errors if the user writes an
+ * invalid predicate.
+ */
+public class AccumulationVisitor extends BaseTypeVisitor<AccumulationAnnotatedTypeFactory> {
+
+    /**
+     * Constructor matching super.
+     *
+     * @param checker the checker
+     */
+    public AccumulationVisitor(BaseTypeChecker checker) {
+        super(checker);
+    }
+
+    /** Checks each predicate annotation to make sure the predicate is well-formed. */
+    @Override
+    public Void visitAnnotation(final AnnotationTree node, final Void p) {
+        AnnotationMirror anno = TreeUtils.annotationFromAnnotationTree(node);
+        if (atypeFactory.isPredicate(anno)) {
+            String errorMessage = atypeFactory.isValidPredicate(anno);
+            if (errorMessage != null) {
+                checker.report(
+                        node,
+                        new DiagMessage(Diagnostic.Kind.ERROR, "predicate.invalid", errorMessage));
+                return null;
```

Comment:
Why not give the superclass the opportunity to issue more errors, if there are others?